### PR TITLE
Video listing

### PR DESCRIPTION
**Pull request by tanveertkd**
This PR contains the video listing page.  
preview-1  
preview-2

### CODE REVIEW

File: src/context/video-listing-context.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            (async () => {
+                try {
+                    const {
+                        data: { videos },
+                        status,
+                    } = await getVidoes();
+                    if (status === 200) videoListDispatch({ type: 'LOAD_VIDEOS', payload: videos });
+                    else throw new Error('Falied with code: ', status);
+                } catch {
+                    throw new Error('Error fetching data.');
```
Comment (BelideNikhil) [inline src/context/video-listing-context.js]: catch will catch error anyway, so you don't need to throw an error explicitly

File: src/components/Sidebar/Sidebar.css [hunk 2]
```diff
@@ -0,0 +1,54 @@
+.nav-sidebar-container {
+    flex: 0 0 250px;
+    flex-direction: column;
+    width: 250px;
```
Comment (BelideNikhil) [inline src/components/Sidebar/Sidebar.css:3]: you can remove this property

**Review by BelideNikhil — APPROVED** [review]
The code is clean and UI looks nice
apart from suggested changes, you are GTG!😁

File: src/context/video-listing-context.js (hunk 1 above)
Comment (tanveertkd) [reply src/context/video-listing-context.js]: Done

File: src/components/Sidebar/Sidebar.css (hunk 2 above)
Comment (tanveertkd) [reply src/components/Sidebar/Sidebar.css:3]: Done

File: src/reducers/video-listing-reducer.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const videoListReducer = (state, action) => {
+    switch(action.type){
```
Comment (AbhishekkGautam) [inline src/reducers/video-listing-reducer.js:2]: we can also maintain states for loading & error here. it'll be easier to implement loader on initial rendering.

File: src/services/videos-services.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import axios from "axios";
+
+const getVidoes = async () => {
+    try{
+        return await axios.get('/api/videos');
+    }catch{
+        console.log('Couldn\'t get vidoes.');
```
Comment (AbhishekkGautam) [inline src/services/videos-services.js:7]: we should remove console.log before commit.

**Review by AbhishekkGautam — APPROVED** [review]
code looks clean & well structured. good to merge

File: src/components/VideoCard/VideoCard.jsx [hunk 5]
```diff
@@ -1,15 +1,23 @@
...
+import { OverflowMenu } from '../OverflowMenu/OverflowMenu';
 
 import './VideoCard.css';
 
 const VideoCard = ({ video }) => {
-
     const reduceTitleLength = (title) => {
-        if(title.length > 50)
-            return title.substring(0, 47)+'...';
+        if (title.length > 50) return title.substring(0, 47) + '...';
```
Comment (himadri2110) [inline src/components/VideoCard/VideoCard.jsx]: Suggestion: You can use ellipsis to trim the title to a specific number of lines.

File: src/App.js [hunk 6]
```diff
@@ -1,20 +1,22 @@
...
+        <div className="App">
+            <div className="section-one">
+                <Navbar />
+            </div>
+            <div className="section-two">
+                <NavigationRoutes />
+            </div>
+            <div className="section-three">
+                <Footer />
+            </div>
```
Comment (himadri2110) [inline src/App.js:17]: The classnames here can be more specific

File: src/pages/VideoListing/VideoListing.jsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+const Videos = () => {
+    const { videoListState } = useVideoProvider();
+    console.log(videoListState)
+
+    return (
+        <div className="video-card-parent">
+            {videoListState.data?.map((video) => (
+                <VideoCard video={video} />
+            ))}
```
Comment (himadri2110) [inline src/pages/VideoListing/VideoListing.jsx]: Suggestion: You can provide a key prop to `VideoCard` to uniquely identify it.

**Review by himadri2110 — APPROVED** [review]
Everything looks good to go.

File: src/reducers/video-listing-reducer.js (hunk 3 above)
Comment (tanveertkd) [reply src/reducers/video-listing-reducer.js:2]: I actually wanted to finish up the basic requirements first, which is why those cases are missing.

File: src/services/videos-services.js (hunk 4 above)
Comment (tanveertkd) [reply src/services/videos-services.js:7]: Noted. Thanks!

File: src/components/VideoCard/VideoCard.jsx (hunk 5 above)
Comment (tanveertkd) [reply src/components/VideoCard/VideoCard.jsx]: Will make the change. Thanks!

File: src/App.js (hunk 6 above)
Comment (tanveertkd) [reply src/App.js:17]: Noted. Thanks!

File: src/pages/VideoListing/VideoListing.jsx (hunk 7 above)
Comment (tanveertkd) [reply src/pages/VideoListing/VideoListing.jsx]: Will make the change.

**Pull request merged**